tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go

```diff
@@ -0,0 +1,229 @@
+				siteconfigOperatorPodName, err := pod.Pull(HubAPIClient, desiredPodName, ranparam.AcmOperatorNamespace)
+				Expect(err).ToNot(HaveOccurred(), "Failed to pull siteconfig operator pod name")
+				if err == nil {
+					_, err = siteconfigOperatorPodName.DeleteAndWait(3 * time.Minute)
+					Expect(err).ToNot(HaveOccurred(), "Failed to delete siteconfig operator pod")
+				}
+			})
+
+			// Teardown test expected results validation.
+			// The default assisted installer template reference ConfigMap custom resource should be recreated successfully.
+			By("checking the default assisted installer template reference ConfigMap CRs recreated successfully.", func() {
+
+				By("checking default assisted installer cluster level templates ConfigMap CR exists")
+				_, err := configmap.Pull(HubAPIClient, tsparams.DefaultAIClusterTemplatesConfigMapName,
+					ranparam.AcmOperatorNamespace)
+				Expect(err).ToNot(HaveOccurred(), "Failed to find default AI cluster level templates config map")
+
+				By("checking default assisted installer node level templates ConfigMap CR exists")
+				_, err = configmap.Pull(HubAPIClient, tsparams.DefaultAINodeTemplatesConfigMapName,
+					ranparam.AcmOperatorNamespace)
+				Expect(err).ToNot(HaveOccurred(), "Failed to find default AI node level templates config map")
+
+				By("verifying installed spoke cluster should still be functional")
+				_, err = version.GetOCPVersion(Spoke1APIClient)
+				Expect(err).ToNot(HaveOccurred(), "Failed to get OCP version from spoke and verify spoke cluster access")
+			})
+		})
+
+		// 75376 - Detaching the AI single-node openshift (SNO) spoke cluster.
+		It("Validate detaching the AI single-node openshift spoke cluster", reportxml.ID("75376"), func() {
+
+			By("checking spoke cluster type")
+			spokeClusterType, err := rancluster.CheckSpokeClusterType(RANConfig.Spoke1APIClient)
+			Expect(err).ToNot(HaveOccurred(), "Failed to fetch spoke cluster type")
+
+			if spokeClusterType == ranparam.StandardCluster {
+				Skip("This test only applies to single-node openshift spoke cluster")
+			}
+
+			// Test step 1-Update the ztp-test git path to delete the ClusterInstance CR in root level kustimozation.yaml.
+			By("updating the Argo CD clusters app with the detach AI SNO cluster instance git path")
+			exists, err := gitdetails.UpdateArgoCdAppGitPath(tsparams.ArgoCdClustersAppName,
+				tsparams.ZtpTestPathDetachAISNO, true)
+			if !exists {
+				Skip(err.Error())
+			}
+
+			Expect(err).ToNot(HaveOccurred(), "Failed to update Argo CD clusters app with new git path")
+
+			// Test step 1 expected results validation.
+			ValidateAISpokeClusterInstallCRsRemoved()
+		})
+	})
+
+// ValidateAISpokeClusterInstallCRsRemoved verifies AI spoke cluster install CRs removed and spoke cluster accessible.
+func ValidateAISpokeClusterInstallCRsRemoved() {
+	By("checking the infra env manifests removed on hub", func() {
```

Comment:
for consistency, could you please switch this PR to use the form of `By` without the `func() {}`?

Reply:
Okay, updated. Thanks!

Reply:
@mpmaruthu I don't see the update here, it looks like there's still a lot that use `func() {}`

Reply:
Yes, I have updated in my local feature branch and validating the changes by running dry-run. Will keep you posted once the new patch is available for review. Thanks!

Reply:
@mpmaruthu Ah that makes sense, I'll be on the lookout